ui/jobs/jobs.js

```diff
@@ -3124,6 +3226,48 @@ class Bars {
           this.buffTracker.onUseAbility(m.groups.id, m.groups);
       }
     }
+    // For extremely complex BRD
+    if (this.job != 'BRD') return;
+    if (!this.dotTarget)
+      this.dotTarget = [];
+    const brdDoTs = Object.freeze([
+      EffectId.Stormbite,
+      EffectId.Windbite,
+      EffectId.CausticBite,
+      EffectId.VenomousBite,
+    ]);
+    if (type === '26') {
+      let m = log.match(kMobGainsEffectFromYouRegex);
+      if (m) {
+        if (Object.values(brdDoTs).includes(m.groups.effectId.toUpperCase()))
+          this.dotTarget.push(m.groups.targetId);
+        let f = this.mobGainEffectFromYouFuncMap[m.groups.effectId.toUpperCase()];
+        if (f)
+          f(name, m.groups);
+      }
+    } else if (type === '30') {
+      let m = log.match(kMobLosesEffectFromYouRegex);
+      if (m) {
+        if (Object.values(brdDoTs).includes(m.groups.effectId.toUpperCase())) {
+          const index = this.dotTarget.indexOf(m.groups.targetId);
+          if (index > -1)
+            this.dotTarget.splice(index, 1);
+        }
+      }
+    } else if (type === '21' || type === '22') {
+      let m = log.match(kYouUseAbilityRegex);
+      if (m) {
+        if (this.dotTarget.includes(m.groups.targetId))
+          this.lastAttackedDotTarget = m.groups.targetId;
+      }
+    } else if (type === '24') {
+      // line[2] is dotted target id.
+      if ((this.dotTarget.includes(line[2])) &&
+        (line[2] === this.lastAttackedDotTarget) &&
```

Comment:
If this is the `lastAttackedDotTarget` then it's already necessarily in `this.dotTarget`, and so `this.dotTarget.includes` shouldn't need to be checked?